### ISSUE TITLE
Investigate using `nanoserde` instead of `serde`

### ISSUE THREAD

**Issue by abdelhamidbakhta** [keep-starknet-strange/kaioshin#64]
# Description

As part of the effort of trying to reduce the build time we might consider replacing `serde` by `nanoserde`.

https://github.com/not-fl3/nanoserde

**Comment by ZePedroResende** [comment]
I would like to tackle this issue

**Comment by ZePedroResende** [comment]
Hey @abdelhamidbakhta after conducting some research on nanoserde and other alternatives like miniserde these are the findings :

## NanoSerde
It was updated last 3 months ago, the API for the serialize and deserialize are different from serde (doesn't return errors on serialize), and doesn't include an `impl_fixed_hash_serde!` macro for example. There are multiple packages that we build that still use serde underneath (sp-runtime, tracing-subscriber, sp-core, etc) so removing it means that we will still be building it.

## Miniserde
It's `JSON` only, and has worst performance (between 1.5 to 2 times) in some use cases (as stated on the [README](https://github.com/dtolnay/miniserde#similar-stupidly-good-performance)).

## TL;DR
Given this constraints our opinion is that we shouldn't move forward with this.

**Comment by abdelhamidbakhta** [comment]
> Hey @abdelhamidbakhta after conducting some research on nanoserde and other alternatives like miniserde these are the findings :
> 
> ## NanoSerde
> It was updated last 3 months ago, the API for the serialize and deserialize are different from serde (doesn't return errors on serialize), and doesn't include an `impl_fixed_hash_serde!` macro for example. There are multiple packages that we build that still use serde underneath (sp-runtime, tracing-subscriber, sp-core, etc) so removing it means that we will still be building it.
> 
> ## Miniserde
> It's `JSON` only, and has worst performance (between 1.5 to 2 times) in some use cases (as stated on the [README](https://github.com/dtolnay/miniserde#similar-stupidly-good-performance)).
> 
> ## TL;DR
> Given this constraints our opinion is that we shouldn't move forward with this.

Ok closing the issue then, thanks
